Makefile

```diff
@@ -283,3 +286,33 @@ COMMIT ?= "HEAD"
 add-tags: | $(MULTIMOD)
 	@[ "${MODSET}" ] || ( echo ">> env var MODSET is not set"; exit 1 )
 	$(MULTIMOD) verify && $(MULTIMOD) tag -m ${MODSET} -c ${COMMIT}
+
+# The source directory for opentelemetry-configuration schema.
+OPENTELEMETRY_CONFIGURATION_JSONSCHEMA_SRC_DIR=tmp/opentelememetry-configuration
+
+# The SHA matching the current version of the opentelemetry-configuration schema to use
+OPENTELEMETRY_CONFIGURATION_JSONSCHEMA_VERSION=v0.1.0
+
+# Cleanup temporary directory
+genjsonschema-cleanup:
+	rm -Rf ${OPENTELEMETRY_CONFIGURATION_JSONSCHEMA_SRC_DIR}
+
+GENERATED_CONFIG=./config/pkg/${OPENTELEMETRY_CONFIGURATION_JSONSCHEMA_VERSION}/generated_config.go
+
+# Generate structs for configuration from opentelemetry-configuration schema
+genjsonschema: genjsonschema-cleanup $(GOJSONSCHEMA)
+	mkdir -p ${OPENTELEMETRY_CONFIGURATION_JSONSCHEMA_SRC_DIR}
+	curl -sSL https://api.github.com/repos/open-telemetry/opentelemetry-configuration/tarball/${OPENTELEMETRY_CONFIGURATION_JSONSCHEMA_VERSION} | tar xz --strip 1 -C ${OPENTELEMETRY_CONFIGURATION_JSONSCHEMA_SRC_DIR}
+	$(GOJSONSCHEMA) \
+		--capitalization ID \
+		--capitalization OTLP \
+		--struct-name-from-title \
+		--package config \
```

Comment:
We could name the package `configmapstructure` if this would be a package for that would help parsing the model using `mapstructure`. Reference: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4373

Reply:
I think we will simply couple this package to some concrete library.